runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/WindowingStrategyTranslationTest.java

```diff
@@ -143,4 +162,142 @@ public void testToProtoAndBackWithComponents() throws Exception {
         proto.getAssignsToOneWindow(),
         equalTo(windowingStrategy.getWindowFn().assignsToOneWindow()));
   }
+
+  private static class CustomWindow extends IntervalWindow {
```

Comment:
Just wondering are you going to add more tests around this `CustomWindow` and `CustomWindowFn` below? It seems like if you only want to test proto-object round trip, there is no need to construct so much details.

Reply:
Yes. This is the one I want to use to test gradually and finally to make custom WindowFn work. Right now it is only private as it is used within this test. In the further as I find more place to add tests I can move this class out accordingly.